extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobNameTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Marco Primo
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DestinationBlobNameTest {
+
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName,  null, 1, blobName",
+            "partName, blobName,  '', 1, blobName",
+    }, nullValues = { "null" })
```

Comment:
given that the changing part is only on a single field (folderName), please change this to `@ValueSource` and `@NullSource`.
Same for the other tests with the same pattern.

Reply:
Would it be better then to separate these ParameterizedTests into two different normal tests? since one test will have the empty folder name and the other null? Because @ValueSource will  only need to have a parameter which is the parameter with an empty string and then I would have to use another test to use the @NullSource annotation

Reply:
Something like 

```
    @Test
    void shouldReturnBlobName_whenFolderNameIsEmptyAndPartSizeIsEqualToOne() {

        String partName="partName", blobName = "blobName",folderName="",expected="blobName";
        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
        assertThat(destinationBlobName.resolve(partName, 1)).isEqualTo(expected);
    }
    @Test
    void shouldReturnBlobName_whenFolderNameIsNullAndPartSizeIsEqualToOne() {

        String partName="partName", blobName = "blobName",folderName=null,expected="blobName";
        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
        assertThat(destinationBlobName.resolve(partName, 1)).isEqualTo(expected);
    }
```

Reply:
you can use `ValueSource` and `NullSource` in the same test, anyway I think separated tests are better in any case.
Just a thought: testing the "null" case is not really needed, because should be implicit with the "empty" one, as the code should always be null-safe and the language offers easy ways to ensure that (e.g. `Optional`)